### PR TITLE
Better error messages from expression evaluation

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: utils/misc.go [hunk 1]
```diff
@@ -20,3 +20,11 @@ func IsNil(v interface{}) bool {
 
 	return false
 }
+
+// MinInt returns the minimum of two integers
+func MinInt(x, y int) int {
```
Comment (rowanseymour) [inline utils/misc.go]: only in go you write your own `min` function...

File: cmd/flowserver/server_test.go [hunk 2]
```diff
@@ -303,7 +303,7 @@ func (ts *ServerTestSuite) TestExpression() {
 	// try the expression endpoint with a missing variable
 	status, body = ts.testHTTPRequest("POST", "http://localhost:8800/expression", `{"expression": "@(foo + 2)", "context": {}}`)
 	ts.Equal(200, status)
-	ts.assertExpressionResponse(body, "", []string{"unable to resolve 'foo'"})
+	ts.assertExpressionResponse(body, "", []string{"error evaluating @(foo + 2): json object has no property 'foo'"})
```
Comment (rowanseymour) [inline cmd/flowserver/server_test.go]: templates can have multiple expressions so error messages now include the expression they came from

Comment (nicpottier) [reply cmd/flowserver/server_test.go]: ooh nice

File: utils/misc.go (hunk 1 above)
Comment (nicpottier) [reply utils/misc.go]: heh

**Pull request merged**